### PR TITLE
Update go-ssz lib dep and usage according to the changes in interface.

### LINKED ISSUE #531: Pull requests should have an associated tracking issue
If you're planning non-trivial work. Be sure to create a github issue prior to starting work. 

That issue should capture background on what the problem is that you are trying to solve and **that issue should be referenced and closed by your pull request**.

### PR DESCRIPTION
**Pull request by katzdan**
Resolves #531 

# Description
Update go-ssz lib dep and usage according to the changes in interface.

**Write why you are making the changes in this pull request**
Code updated to use the Marshal/Unmarshal function of the go-ssz lib instead of 'Encode' and 'Decode'.

**Write a summary of the changes you are making**
1. Updated the ssz dependency to the latest go-ssz commit hash (as of now).
2. Changed the calls to ssz lib : ssz.Encode -> ssz.Marshal , ssz.Decode -> ssz.Unmarshal

**Link anything that would be helpful or relevant to the reviewers**
Currently the call to Unmashal results in panic. I need someone to help figure out if this is a go-ssz lib bug or something is wrong in my code.

### CODE REVIEW

**Comment by katzdan** [comment]
@prestonvanloon , 
Unmarshaling fails with DepositInput and from what @nisdas - there are still bugs to be fixed in the go-ssz lib. I suggest to continue testing the code changes after go-ssz has reached a stable version.
Your thoughts?

**Comment by prestonvanloon** [comment]
Blocked by https://github.com/prysmaticlabs/go-ssz/issues/32

File: WORKSPACE [hunk 1]
```diff
@@ -154,16 +154,16 @@ load("@com_github_bazelbuild_buildtools//buildifier:deps.bzl", "buildifier_depen
...
-    sha256 = "f6fd5d623a988337810b956ddaf612dce771d9d0f9256934c8f4b1379f1cb2f6",
-    strip_prefix = "go-ssz-2e84733edbac32aca6d47feafc4441e43b10047f",
-    url = "https://github.com/prysmaticlabs/go-ssz/archive/2e84733edbac32aca6d47feafc4441e43b10047f.tar.gz",
-)
+# http_archive(
+#     name = "com_github_prysmaticlabs_go_ssz",
+#     sha256 = "f6fd5d623a988337810b956ddaf612dce771d9d0f9256934c8f4b1379f1cb2f6",
+#     strip_prefix = "go-ssz-2e84733edbac32aca6d47feafc4441e43b10047f",
+#     url = "https://github.com/prysmaticlabs/go-ssz/archive/2e84733edbac32aca6d47feafc4441e43b10047f.tar.gz",
+# )
```
Comment (prestonvanloon) [inline WORKSPACE]: This can be a go_repository, but it needs to call `go_ssz_dependencies` too.

Please remove commented code.

Comment (katzdan) [reply WORKSPACE]: ok

Comment (katzdan) [reply WORKSPACE]: done.

**Comment by katzdan** [comment]
@prestonvanloon - how should I proceed with this PR?

**Comment by katzdan** [comment]
After trying with the go-ssz (crrently) latest commit hash - 

I'm still getting errors:
[2019-07-15 07:04:37]  INFO sync: Starting service
[2019-07-15 07:04:37]  INFO rpc: Starting service
[2019-07-15 07:04:37]  INFO rpc: Listening on port port=4000
[2019-07-15 07:04:37]  WARN rpc: You are using an insecure gRPC connection! Provide a certificate and key to connect securely
[2019-07-15 07:04:37]  INFO prometheus: Starting service endpoint=:8080

panic: runtime error: slice bounds out of range

goroutine 220 [running]:
github.com/prysmaticlabs/go-ssz.makeStructUnmarshaler.func1(0xc00058a490, 0xc0, 0x110, 0x133dca0, 0xc000edae00, 0x199, 0x0, 0x22c5f90, 0x1380c00, 0xc0003a1a80)
	external/com_github_prysmaticlabs_go_ssz/unmarshal.go:388 +0xe23
github.com/prysmaticlabs/go-ssz.makePtrUnmarshaler.func1(0xc00058a490, 0xc0, 0x110, 0x1380c00, 0xc0000111a0, 0x196, 0x0, 0x300, 0x2ae, 0x0)
	external/com_github_prysmaticlabs_go_ssz/unmarshal.go:407 +0x99
github.com/prysmaticlabs/go-ssz.Unmarshal(0xc00058a490, 0xc0, 0x110, 0x1164d80, 0xc0000111a0, 0xc0003a1020, 0x196)
	external/com_github_prysmaticlabs_go_ssz/unmarshal.go:39 +0x38d
github.com/prysmaticlabs/prysm/beacon-chain/core/helpers.DecodeDepositInput(0xc00058a480, 0xd0, 0x120, 0x9e834bc77e14bb58, 0x85407cf610805629, 0xa13529f24ebe93d4)
	beacon-chain/core/helpers/deposits.go:56 +0x2b2
github.com/prysmaticlabs/prysm/beacon-chain/powchain.(*Web3Service).ProcessDepositLog(0xc000025080, 0x9c704d2847bca002, 0xadcd662278f7db31, 0xb471a6a, 0xc0000a2000, 0x1, 0x4, 0xc00058a000, 0x5a0, 0x5a0, ...)
	beacon-chain/powchain/log_processing.go:80 +0x171
github.com/prysmaticlabs/prysm/beacon-chain/powchain.(*Web3Service).ProcessLog(0xc000025080, 0x9c704d2847bca002, 0xadcd662278f7db31, 0xb471a6a, 0xc0000a2000, 0x1, 0x4, 0xc00058a000, 0x5a0, 0x5a0, ...)
	beacon-chain/powchain/log_processing.go:48 +0x41f
github.com/prysmaticlabs/prysm/beacon-chain/powchain.(*Web3Service).processPastLogs(0xc000025080, 0xc0000b3b00, 0x31a377574ff1c37)
	beacon-chain/powchain/log_processing.go:175 +0x20c
github.com/prysmaticlabs/prysm/beacon-chain/powchain.(*Web3Service).run(0xc000025080, 0xc000481980)
	beacon-chain/powchain/service.go:326 +0x460
created by github.com/prysmaticlabs/prysm/beacon-chain/powchain.(*Web3Service).Start
	beacon-chain/powchain/service.go:165 +0x1a7

@prestonvanloon  - please open this PR.

**Pull request closed without merging**